virtual_database_client/path.go

```diff
@@ -0,0 +1,329 @@
+package client
+
+import (
+	"fmt"
+	"strconv"
+
+	"encoding/json"
+	"net"
+	"regexp"
+	"time"
+
+	spb "github.com/Azure/sonic-telemetry/proto"
+	"github.com/go-redis/redis"
+	log "github.com/golang/glog"
+	gnmipb "github.com/openconfig/gnmi/proto/gnmi"
+)
+
+// gnmiFullPath builds the full path from the prefix and path.
+func gnmiFullPath(prefix, path *gnmipb.Path) *gnmipb.Path {
+
+	fullPath := &gnmipb.Path{Origin: path.Origin}
+	if path.GetElement() != nil {
+		fullPath.Element = append(prefix.GetElement(), path.GetElement()...)
+	}
+	if path.GetElem() != nil {
+		fullPath.Elem = append(prefix.GetElem(), path.GetElem()...)
+	}
+	return fullPath
+}
+
+func populateAlltablePaths(prefix *gnmipb.Path, paths []*gnmipb.Path, pathG2S *map[*gnmipb.Path][]tablePath) error {
+	for _, path := range paths {
+		err := populateNewtablePath(prefix, path, pathG2S)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// First translate gNMI paths to a list of unique
+// root-to-leaf virtual paths in the vpath tree.
+// Then map each vpath to a list of redis DB path.
+func populateNewtablePath(prefix, path *gnmipb.Path, pathG2S *map[*gnmipb.Path][]tablePath) error {
+	//fullPath := path
+	//if prefix != nil {
+	//	fullPath = gnmiFullPath(prefix, path)
+	//}
+	//fmt.Printf("fullPath: %v\n", fullPath)
+
+	target := prefix.GetTarget()
+	stringSlice := []string{target}
+	elems := path.GetElem()
+	for _, elem := range elems {
+		stringSlice = append(stringSlice, elem.GetName())
+	}
+
+	err := searchPathTrie(stringSlice, path, pathG2S)
+	if err != nil {
+		return err
+	}
+
+	// Debug
+	//for k, v := range (*pathG2S) {
+	//	fmt.Printf("gNMI path: %v\n", k)
+	//	for _, tblPath := range v{
+	//		fmt.Printf("new tablePath: %v\n", tblPath)
```

Comment:
Convert to log message and uncomment or remove.

Reply:
Removed.